### DISCUSSION TITLE
How to get the vertices from Hatch?

### DISCUSSION THREAD

**Discussion by derekcbr** [mozman/ezdxf#347] [Q&A]
```
if entity.dxftype()=='HATCH':
            print([(attrib, entity.dxf.get(attrib)) for attrib in entity.dxf.all_existing_dxf_attribs()])
```
It only prints as below:
[('handle', '105'), ('owner', '70'), ('layer', 'A-WALL'), ('color', 8), ('elevation', Vec3(0.0, 0.0, 0.0)), ('extrusion', Vec3(0.0, 0.0, 1.0)), ('pattern_name', 'SOLID'), ('solid_fill', 1), ('associative', 0), ('hatch_style', 0), ('pattern_type', 1)]

How to get the vertices from Hatch?

**Comment by mozman** [comment]
Its not that easy in the current stable version v0.15, but it is easy with the next version v0.16 or the head of the master branch:

I show you only the easy way for v0.16, for v0.15 you would have to deal with LWPOLYLINE paths and/or LINE, ARC, ELLIPSE and SPLINE edge paths:

```Python
from ezdxf import disassemble

# HATCH may have multiple boundary paths and each is a separated Primitive() object:
primitives = disassemble.to_primitives([entity], max_flattening_distance=0.01)
for p in primitives:
    # all curves get approximated:
    vertices = list(p.vertices())
    # and process the vertices of one boundary path

```
The `max_flattening_distance` defines the max distance between the  approximation line and the original curve. Use argument
`max_flattening_distance` to control the approximation accuracy.
